pkg/host/store.go

```diff
@@ -163,11 +163,6 @@ func (s *Store[E]) Delete(ctx context.Context, id string) error {
 		return fmt.Errorf("failed to set object metadata: %w", err)
 	}
 
-	s.enqueue(store.WatchEvent[E]{
-		Type:   store.WatchEventTypeDeleted,
-		Object: obj,
-	})
-
```

Comment:
Is there a reason for deleting this?

Reply:
Since the garbage collector now fully manages the deletion process, there's no need to enqueue reconciliation. Setting the DeleteAt field alone is now adequate for gracefully shutting down the machine.